### PR TITLE
Jetpack Connect: Move flow types constants to the section

### PR DESCRIPTION
**Pull request by tyxla**
#45437 introduces some flow type constants to the Jetpack Connect state constants file. Due to that, and the fact that they're using `shouldShowOfferResetFlow`, that caused a significant 12% increase of the main entry point bundle. That was discovered by @sgomes today.

However, since those flow types aren't used at all in the state code, and they're only used in components, we're moving them to the Jetpack Connect section. Furthermore, because they import a bunch of code in their dependencies, we're intentionally moving them to their own constants file, so they'll be used only if needed, and that way we can optimize for the best webpack heuristics possible.

We're also changing the `shouldShowOfferResetFlow` function to a simple boolean flag, because it was pulling `lib/abtest`, and that pulls a whole lot of additional dependencies, just to check something that is already 100% true (that A/B test is currently running the same variant for 100% of the users).

So, if you look at the changes in the bundles, you'll notice a close resemblance to what would have been a revert of #45437. 

A good next step would be to remove all the code that isn't used anymore since the Offer Reset is open to 100% of the users. That would remove not only the `shouldShowOfferResetFlow` flag but also a bunch of other components and possibly state.

#### Changes proposed in this Pull Request

* Move Jetpack flow types to their own file.
* Remove abtest getters completely and replace them with a simple flag.

#### Testing instructions

* Go through the testing instructions of #45437.

##### Notes

Lint errors are expected.

### CODE REVIEW

**Review by wp-desktop — DISMISSED** [review]
WordPress Desktop CI Failure for job "wp-desktop-mac".

@tyxla please inspect this job's build steps for breaking changes at [this link](https://circleci.com/gh/Automattic/wp-calypso/895644). For temporal failures, you may try to "Rerun Workflow from Failed".

Please also ensure this branch is rebased off latest Calypso.

File: client/my-sites/plans/test/index.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ jest.mock( 'my-sites/controller', () => ( {
 	siteSelection: jest.fn(),
 	sites: jest.fn(),
 } ) );
-jest.mock( 'lib/abtest/getters', () => ( {
+jest.mock( 'lib/plans/config', () => ( {
```
Comment (tyxla) [inline client/my-sites/plans/test/index.js:21]: Noting that I'd prefer to completely get rid of these tests, as I don't really see the value in having them.

Comment (rcanepa) [reply client/my-sites/plans/test/index.js:21]: I think they had value during development when we needed to be absolutely sure that only users in the experiment were seeing the new plans page but now they don't serve any purpose.

Once we know for sure we are not reverting anything back, we will remove the `shouldShowOfferResetFlow` check from everywhere and remove these tests as well. How that sounds to you?

**Review by rcanepa — APPROVED** [review]
LGTM. Code looks good and the app works as expected.

@tyxla thanks a lot for taking care of this. It's clear to me now what caused the increase in the bundle size. I'll make sure to share what happened here with the rest of the team to avoid repeating it in the future.

**Review by sgomes — COMMENTED** [review]
The code and the ICFY results look good to me as well. Great work, @tyxla! 👍

**Review by sgomes — APPROVED** [review]
Forgot to approve 😄

**Comment by tyxla** [comment]
Thanks, folks 🙌

**Pull request merged**